### PR TITLE
fix(repos): Handle integrity error in repo creation

### PR DESCRIPTION
**Pull request by macqueen**
fixes SENTRY-3BM

### CODE REVIEW

**Comment by MaxBittker** [comment]
don't quite understand, if there was an integrity error, the creation failed, so why do we need to delete it? is this to delete the row that may have been there beforehand and caused the integrity error?

File: src/sentry/plugins/providers/repository.py [hunk 1]
```diff
@@ -65,14 +66,34 @@ def dispatch(self, request, organization, **kwargs):
...
+            try:
+                repo = Repository(
+                    organization_id=organization.id,
+                    name=result['name'],
+                    external_id=result.get('external_id'),
+                    url=result.get('url'),
+                    config=result.get('config') or {},
+                    provider=self.id,
+                )
+                self.delete_repository(repo, actor=request.user)
```
Comment (macqueen) [inline src/sentry/plugins/providers/repository.py]: @MaxBittker kind of confusing, but this `self.delete_repository` doesn't actually delete anything from the database, it just deletes the webhook that we would have already created in github/bitbucket (see line 55 above: https://github.com/getsentry/sentry/pull/6411/files#diff-7158a0c29ea647b04fbf0562adb8baccR55)

Comment (MaxBittker) [reply src/sentry/plugins/providers/repository.py]: oh!

**Review by MaxBittker — APPROVED** [review]
lgtm 👍

**Pull request merged**